### PR TITLE
Add dataset guide

### PR DESCRIPTION
**Pull request by KennethEnevoldsen**
Dear Peters,

Added a guide for adding a dataset here. Let me know if anything is unclear.

### CODE REVIEW

File: docs/Adding_a_new_dataset [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+   2) Languages (e.g. "da" for Danish)
+2) Add a dataset of the same name to the `/danish-foundation-models (193701)/dfm-data/pre-training/` on UCloud. Using the following folder structure:
+
+```
+pre-training
+│
+└── dataset_name
+    │
+    ├── documents
+    │   └── dataset_name.jsonl.gz
```
Comment (peterbjorgensen) [inline docs/Adding_a_new_dataset:16]: The jsonl files do not need to have the same name as `dataset_name`, and this part of the documentation implies that it is a requirement. Also, it's recommended that the documents are split across multiple files, which makes parallel data processing easier (approximately 1M documents or around 1 GB per file is a sensible tradeoff), so I think it's better if the example shows `part1.jsonl.gz`, `part2.jsonl.gz` etc.

**Review by peterbjorgensen — CHANGES_REQUESTED** [review]
Looks good to me, except for the minor comment.

File: docs/Adding_a_new_dataset (hunk 1 above)
Comment (KennethEnevoldsen) [inline docs/Adding_a_new_dataset:16]:
```suggestion
    │   ├──part1.jsonl.gz
    │   ├──part2.jsonl.gz
    │   └── ...
```

Comment (KennethEnevoldsen) [reply docs/Adding_a_new_dataset]: @TTTTao725 just to make sure you see this

**Review by peterbjorgensen — APPROVED** [review]

**Pull request merged**